physx/src/scene.rs

```diff
@@ -88,6 +91,39 @@ impl Scene {
         })
     }
 
+    pub fn add_capsule_controller(
+        &mut self,
+        material: *mut PxMaterial,
+        height: f32,
+        radius: f32,
+        step_offset: f32,
+    ) -> Controller {
+        unsafe {
+            if self.controller_manager.is_none() {
+                self.controller_manager = Some(phys_PxCreateControllerManager(
+                    self.px_scene.write().unwrap().expect("accessing null ptr"),
+                    false,
+                ));
+            }
+
+            let c = PxCapsuleControllerDesc_new_alloc();
```

Comment:
If you're up for it, I'd prefer if this was an actual rust struct as well, so we stick closer to the PhysX api. That also makes it easier if we end up exposing things like Materials as Rust-level types. Then we can also get rid of the panic, and let the caller decide what to do if the struct wasn't constructed correctly.

If you know, I'm also curious about the ownership here - this returns a pointer, but it's not clear to me if the controller or controller manager takes ownership of it later.

Reply:
Looking at it again, I don't think either take ownership - they just read the data. So it should be ok to delete it straight after using it.

Reply:
Ok, I had a go at converting into proper rust struct. Let me know what you think!